Extend/Controller/OrderController.php

```diff
@@ -246,14 +316,43 @@ private function _handleResponse($oResponse, $oLogger)
                 $sDescription = $oStatus->getDescription();
                 $oLogger->error("\t$sSeverity with code $sCode and message '$sDescription' occurred.");
             }
-            return;
+            return false;
+        }
+
+        $oOrder->oxorder__wdoxidee_orderstate = new Field(Order::STATE_PENDING);
+        $oOrder->oxorder__wdoxidee_transactionid = new Field($oResponse->getTransactionId());
+        $oOrder->save();
+
+        if ($oResponse instanceof FormInteractionResponse) {
+            /**
+             * @var $oSession \OxidEsales\Eshop\Core\Session
+             */
+            $oSession = Registry::getSession();
+            $oSession->setVariable(
+                "wdFormInteractionResponse",
+                new FormInteractionResponseFields(
+                    $oResponse->getUrl(),
+                    $oResponse->getMethod(),
+                    $oResponse->getFormFields()
+                )
+            );
+
+            $sSid = $oSession->sid(true);
+            if ($sSid != '') {
+                $sSid = '&' . $sSid;
```

Comment:
You can also use `http_build_query` for this.

Reply:
no, `$sSid` is already a query